### PR TITLE
bug compatible with `stdin.on("readable")`

### PR DESCRIPTION
**Pull request by alexlamsl**
- fix variable name typo
- fix tests

### CODE REVIEW

File: src/bun.js/builtins/js/ProcessObjectInternals.js [hunk 1]
```diff
@@ -509,20 +510,38 @@ function getStdinStream(fd, rawRequire, Bun) {
...
     }
 
     resume() {
-      this.#reader ??= Bun.stdin.stream().getReader();
       this.ref();
       return super.resume();
     }
 
     ref() {
+      this.#reader ??= Bun.stdin.stream().getReader();
```
Comment (Jarred-Sumner) [inline src/bun.js/builtins/js/ProcessObjectInternals.js:541]: There's supposed to be a way to ref() it directly from the stream. It should auto-unref when stdin itself closes. But, that might lead to a race condition in this case

**Review by Jarred-Sumner — APPROVED** [review]

File: src/bun.js/builtins/js/ProcessObjectInternals.js (hunk 1 above)
Comment (alexlamsl) [reply src/bun.js/builtins/js/ProcessObjectInternals.js:541]: The original code uses `Readable.fromWeb(Bun.stdin.stream())`, and that wrapper class does implement `.[un]ref()`:
https://github.com/oven-sh/bun/blob/2b622162ef6fc999ece19a9bbbc265af326b84a7/src/bun.js/streams.exports.js#L5857-L5858
https://github.com/oven-sh/bun/blob/2b622162ef6fc999ece19a9bbbc265af326b84a7/src/bun.js/streams.exports.js#L6093-L6113

However it doesn't do what I'd expect it to, i.e. preventing the process from exiting:
```js
$ cat test.js
var stdin = Bun.stdin.stream();
var stream = require("stream").Readable.fromWeb(stdin);
console.log(stream.ref, stream.unref);
stream.ref();
stream.on("data", console.log);
// deliberating not calling `.unref()`
// expect process to stick around
```
Typed `abc<Enter>`
```console
$ bun test.js
[Function: ref] [Function: unref]
abc
$
```

**Comment by Jarred-Sumner** [comment]
After rebuilding builtins, this test fails due to `data` being empty:
```js
  it("should allow us to read from stdin in readable mode", (done) => {
    const input = "hello\n";
    // Child should read from stdin and write it back
    const child = spawn("bun", [CHILD_PROCESS_FILE, "STDIN", "READABLE"]);
    let data = "";
    child.stdout.setEncoding("utf8");
    child.stdout
      .on("data", (chunk) => {
        data += chunk;
      })
      .on("end", function () {
        try {
          expect(data).toBe(`data: ${input}`);
          done();
        } catch (err) {
          done(err);
        }
      });
    child.stdin.write(input);
    child.stdin.end();
  });

```

**Comment by alexlamsl** [comment]
That is curious &minus; after `bun upgrade --canary` just now I can run `bun wiptest` without any failures:
```console
$ bun wiptest process-stdio
bun wiptest v0.3.1 (79abd245)

process-stdio.test.ts:
✓ process.stdin
✓ process.stdin - read
✓ process.stdin - resume
✓ process.stdout
✓ process.stderr
✓ process.stdout - write
✓ process.stdout - write a lot (string)
✓ process.stdout - write a lot (bytes)

child-process-stdio.test.js:
✓ process.stdout > should allow us to write to it
✓ process.stdin > should allow us to read from stdin in readable mode
✓ process.stdin > should allow us to read from stdin via flowing mode
✓ process.stdin > should allow us to read > 65kb from stdin
✓ process.stdin > should allow us to read from a file

 13 pass
 0 fail
 22 expect() calls
 Ran 13 tests across 2 files [1.59s]
```
This is on VirtualBox running Ubuntu 22.04:
```console
$ uname -mprs
Linux 5.15.0-56-generic x86_64 x86_64
```

File: src/bun.js/builtins/js/ProcessObjectInternals.js (hunk 1 above)
Comment (alexlamsl) [reply src/bun.js/builtins/js/ProcessObjectInternals.js:541]: IMHO if we resolve #1607 all these `.ref()` here or in the underlying `streams` implementations would no longer be necessary, as there should be no additional keep-alive requirements in addition to `bun`/Web API.

Unless of course I misunderstood the purpose of `.ref()`, as I infer it from [Node.js Timers](https://nodejs.org/dist/latest-v18.x/docs/api/timers.html#timeoutref)

**Pull request merged**